Tasks/Rabushenka/Tasks7/HomWr-7/episode_bl.py

```diff
@@ -0,0 +1,108 @@
+import episode_data
+
+def get_all_1():
+    try:
+        product = episode_data.get_all_1()
+        list_product = " < Available products in the store > \n"
+        for x in product:
+            list_product += f"Product: {x[0]}; price - {x[1]}$; In stock:{x[2]}; Product code:({x[3]}). \n"
+        return list_product
+    except:
+        return "<\nError!> Try again.\n"
+
+'''get_all_2 = factory(episode_data.get_all_2) and it was possible to write with the help of a factory?'''
+
+def get_all_2():
+    try:
+        product = episode_data.get_all_2()
+        list_product = " < Available products in the store > \n"
+        for x in product:
+            list_product += f"Product: {x[0]}; price - {x[1]}$; In stock:{x[2]}; Product code:({x[3]})."'\n'
+        return list_product
+    except:
+        return "<\nError!> Try again.\n"
+
+
+def get_all_3():
+    try:
+        product = episode_data.get_all_3()
+        list_product = "< Available products in the store >\n"
+        for x in product:
+            list_product += f"Product: {x[0]}; price - {x[1]}$; In stock:{x[2]}; Product code:({x[3]})."'\n'
+        return list_product
+    except:
+        return "<\nError!> Try again.\n"
+
+
+def get_all_4():
+    try:
+        product = episode_data.get_all_4()
+        list_product = "< Available products in the store >\n"
+        for x in product:
+            list_product += f"Product: {x[0]}; price - {x[1]}$; In stock:{x[2]}; Product code:({x[3]})."'\n'
+        return list_product
+    except:
+        return "<\nError!> Try again.\n"
```

Comment:
this is a huge load of copy-paste code, which should be replaced with some generic method to get any category